src/main/ruby/core/main.rb

```diff
@@ -42,6 +42,11 @@ def include(*mods)
     end
   end
 
+  def using(mod)
+    Truffle.invoke_primitive(:main_using, mod)
+    self
```

Comment:
Why a primitive and not a CoreMethod if the Ruby parts only calls it?

Reply:
There's some exception logic here that needs to be added:
https://github.com/ruby/ruby/blob/8dbc8caa0dfa366af6818b04e3ee4aebfe09478d/eval.c#L1467
I'm not sure yet how to do this top level check correctly.  It will probably make sense to move this to a core node.

Reply:
I've updated this to use CoreMethod